File: Problems/0475-heaters.py
class Solution(object):
    def findRadius(self, houses, heaters):
        houses.sort()
        heaters.sort()

        ans = 0
        j = 0
        m = len(heaters)

        for house in houses:
            while j + 1 < m and abs(heaters[j + 1] - house) <= abs(heaters[j] - house):
                j += 1

            d = abs(heaters[j] - house)
            if d > ans:
                ans = d

        return ans

File: Problems/0475-heaters.cpp
#include <bits/stdc++.h>
using namespace std;

// LeetCode solution starts from here
class Solution {
public:
    int findRadius(vector<int>& houses, vector<int>& heaters) {
        sort(houses.begin(), houses.end());
        sort(heaters.begin(), heaters.end());

        int ans = 0, j = 0, m = heaters.size();

        for (int house : houses) {
            while (j + 1 < m &&
                   abs(heaters[j + 1] - house) <= abs(heaters[j] - house))
                j++;

            ans = max(ans, abs(heaters[j] - house));
        }

        return ans;
    }
};